File: cinema-backend/src/main/java/org/example/cinemabackend/cinema/application/dto/response/ScreeningResponse.java
package org.example.cinemabackend.cinema.application.dto.response;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Builder;
import org.example.cinemabackend.movie.application.dto.response.MovieResponse;
import org.example.cinemabackend.movie.application.dto.response.MovieVariantResponse;

import java.time.LocalDateTime;

@Builder
public record ScreeningResponse(
        Long id,
        MovieResponse movie,
        @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm") LocalDateTime startTime,
        @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm") LocalDateTime endTime,
        ScreeningRoomResponse screeningRoom,
        MovieVariantResponse movieVariant
) {
}

File: cinema-customer-frontend/src/app/repertory/dtos/response/screening.response.ts
import {ScreeningRoomResponse} from "./screening-room.response";
import {MovieResponse} from "./movie.response";
import {MovieVariantResponse} from "./movie-variant.response";

export class ScreeningResponse {
  constructor(
    readonly id: number,
    readonly movie: MovieResponse,
    readonly startTime: Date,
    readonly endTime: Date,
    readonly screeningRoom: ScreeningRoomResponse,
    readonly movieVariant: MovieVariantResponse
  ) {
  }
}
